### PR TITLE
[win32][fix] Quick test to fix one addon installer issue

### PR DESCRIPTION
**Pull request by Paxxi**
This maybe can fix the addon installer issue? Or maybe not, doesn't seem to have broken my local test so yay for that

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/FilesystemInstaller.cpp [hunk 1]
```diff
@@ -52,18 +52,36 @@ bool CFilesystemInstaller::InstallToFilesystem(const std::string& archive, const
...
+      }
+    } while (!result && count++ < 4);
+
+    if (!result)
       return false;
-    }
   }
 
-  if (!CFile::Rename(newAddonData, addonFolder))
+  auto count = 1;
```
Comment (thebrid) [inline xbmc/addons/FilesystemInstaller.cpp]: These 12 lines are near-duplicates of the 12 lines starting on line 55. Consider refactoring them into a function.

Comment (Paxxi) [inline xbmc/addons/FilesystemInstaller.cpp]: They are but slightly different so I decided against making it a function as it wouldn't be much better for just these two places

Comment (thebrid) [inline xbmc/addons/FilesystemInstaller.cpp]: The 2 pieces perform the same operation on different inputs. There are a lot of advantages to eliminating duplication. If someone finds a bug in one, they may correct it there but not in the other. Similarly if someone makes an enhancement - they may apply it in one copy but not the other.

Comment (MartijnKaijser) [inline xbmc/addons/FilesystemInstaller.cpp]: @thebrid if you are willing to do so feel free to make a PR. We have many areas that need work and we lack enough developers.

Comment (thebrid) [inline xbmc/addons/FilesystemInstaller.cpp]: Coming up :smile_cat:

File: xbmc/addons/FilesystemInstaller.cpp [hunk 2]
```diff
@@ -52,18 +52,36 @@ bool CFilesystemInstaller::InstallToFilesystem(const std::string& archive, const
...
-    if (!CFile::Rename(addonFolder, oldAddonData))
+    auto count = 1;
+    auto result = false;
+    do
     {
-      CLog::Log(LOGERROR, "Failed to move old addon files from '%s' to '%s'", addonFolder.c_str(), oldAddonData.c_str());
+      result = CFile::Rename(addonFolder, oldAddonData);
+      if (!result)
+      {
+        CLog::Log(LOGERROR, "Failed to move old addon files from '%s' to '%s', reytring in 500ms", addonFolder.c_str(), oldAddonData.c_str());
```
Comment (thebrid) [inline xbmc/addons/FilesystemInstaller.cpp]: Also typo here, will fix.

File: xbmc/addons/FilesystemInstaller.cpp [hunk 3]
```diff
@@ -52,18 +52,36 @@ bool CFilesystemInstaller::InstallToFilesystem(const std::string& archive, const
...
+        Sleep(500);
```
Comment (thebrid) [inline xbmc/addons/FilesystemInstaller.cpp]: Build is complaining of missing declaration for Sleep(). Should this be including threads/Thread.h?

File: xbmc/addons/FilesystemInstaller.cpp (hunk 1 above)
Comment (thebrid) [inline xbmc/addons/FilesystemInstaller.cpp]: @Paxxi , sent a PR to your branch.

File: xbmc/addons/FilesystemInstaller.cpp (hunk 3 above)
Comment (FernetMenta) [inline xbmc/addons/FilesystemInstaller.cpp]: XTimeUtils.h for posix

Comment (thebrid) [inline xbmc/addons/FilesystemInstaller.cpp]: :+1:

**Comment by Paxxi** [comment]
jenkins build this please

**Pull request merged**